File: src/main/java/io/zeitwert/ddd/doc/model/DocMeta.java

package io.zeitwert.ddd.doc.model;

import java.util.List;

import io.zeitwert.ddd.aggregate.model.AggregateMeta;
import io.zeitwert.ddd.doc.model.enums.CodeCaseDef;
import io.zeitwert.ddd.doc.model.enums.CodeCaseStage;

public interface DocMeta extends AggregateMeta {

	List<DocPartTransition> getTransitionList();

	CodeCaseDef getCaseDef();

	List<CodeCaseStage> getCaseStages();

	// availableActions: string[];

}

File: fm-dddrive/src/main/java/io/dddrive/doc/model/DocMeta.kt
package io.dddrive.doc.model

import io.dddrive.ddd.model.AggregateMeta
import io.dddrive.doc.model.enums.CodeCaseDef
import io.dddrive.doc.model.enums.CodeCaseStage
import java.time.OffsetDateTime

interface DocMeta : AggregateMeta {

	val docTypeId: String

	val caseDef: CodeCaseDef?

	val caseStages: List<CodeCaseStage>

	val caseStage: CodeCaseStage?

	fun setCaseStage(
		caseStage: CodeCaseStage,
		userId: Any,
		timestamp: OffsetDateTime,
	)

	val isInWork: Boolean

	// availableActions: string[];

	val transitionList: List<DocPartTransition>

}
